docs/command-reference.md

```diff
@@ -0,0 +1,632 @@
+    *   `--debug`: enable debug logging
+    *   `-h, --help`: show help
+
+**Examples**
+
+```bash
+# run a container and attach an interactive shell
+container run -it ubuntu:latest /bin/bash
+
+# run a background web server
+container run -d --name web -p 8080:80 nginx:latest
+
+# set environment variables and limit resources
+container run -e NODE_ENV=production --cpus 2 --memory 1G node:18
+```
+
+### `container build`
+
+Builds an OCI image from a local build context. It reads a Dockerfile (default `Dockerfile`) and produces an image tagged with `-t` option. The build runs in isolation using BuildKit, and resource limits may be set for the build process itself.
+
+**Usage**
+
+```bash
+container build [OPTIONS] PATH
+```
+
+**Options**
+
+*   **Resource management**
+    *   `-c, --cpus <number>`: CPUs to allocate to the build process (default 2)
+    *   `-m, --memory <size>`: memory for the build process (default 2048MB)
+*   **Build configuration**
+    *   `--build-arg <key=value>`: build-time variables passed to the Dockerfile
+    *   `-f, --file <path>`: path to the Dockerfile (default `Dockerfile`)
+    *   `-l, --label <key=value>`: add metadata labels to the image
+    *   `--no-cache`: disable cache usage
+    *   `-o, --output <config>`: specify build output (default `type=oci`)
+    *   `--arch <arch>`: target architecture (default `arm64`)
+    *   `--os <os>`: target operating system (default `linux`)
+    *   `--progress <type>`: progress output mode: `auto`, `plain`, or `tty`
+    *   `--vsock-port <port>`: vsock port used by BuildKit (default 8088)
+    *   `-t, --tag <name>`: set image name and tag
+    *   `--target <stage>`: set the target stage for multi-stage builds
+    *   `-q, --quiet`: suppress build output
+*   **Hidden options**
+    *   `--cache-in <config>`: configure build cache imports
+    *   `--cache-out <config>`: configure build cache exports
+*   **Global**
+    *   `--debug`: enable debug logging
+    *   `-h, --help`: show help
+
+**Examples**
+
+```bash
+# build an image and tag it as my-app:latest
+container build -t my-app:latest .
+
+# use a custom Dockerfile
+container build -f docker/Dockerfile.prod -t my-app:prod .
+
+# pass build args
+container build --build-arg NODE_VERSION=18 -t my-app .
+
+# build the production stage only and disable cache
+container build --target production --no-cache -t my-app:prod .
+```
+
+## Container Management
+
+### `container create`
+
+Creates a container from an image without starting it. This command accepts most of the same process/resource/management flags as `container run`, but leaves the container stopped after creation.
+
+**Usage**
+
+```bash
+container create [OPTIONS] IMAGE [COMMAND] [ARG...]
+```
+
+**Typical use**: create a container to inspect or modify its configuration before running it.
+
+### `container start`
+
+Starts one or more stopped containers. You can attach to the container's output streams and optionally keep STDIN open.
```

Comment:
I don't think container start supports specifying multiple containers to start in the same command